client/components/theme/index.jsx

```diff
@@ -172,8 +176,12 @@ const Theme = React.createClass( {
 								context: 'singular noun, the currently active theme'
 							} ) }</span>
 						}
-						{ price && ! purchased &&
-							<span className="theme-badge__price">{ price }</span>
+						{
+							config.isEnabled( 'jetpack/pijp' )
+							? ! this.props.isJetpack && price && ! purchased &&
```

Comment:
Adding this special case here seems wrong. If we see this component as an abstraction of a theme (check e.g. the other props, they're all supposed to be pretty theme specific), it shouldn't really care what site it is installed on. Also, there are other situations where we want to hide the price, which we're handling thru the `purchased` field (e.g. premium WP.com plan), which I think is appropriate in this case too -- after all that's what we're trying to express: if the theme has been purchased (be it via individual theme purchase, or thru a plan), hide its price.